### PR TITLE
[201911][config] Support abbreviation

### PR DESCRIPTION
**Pull request by jleveque**
Backport https://github.com/Azure/sonic-utilities/pull/893 to the 201911 branch

### CODE REVIEW

File: config/main.py [hunk 1]
```diff
@@ -62,6 +62,44 @@ def log_error(msg):
...
+        shortest = None
+        for x in self.list_commands(ctx):
+            if x.lower().startswith(cmd_name.lower()):
+                matches.append(x)
+                if not shortest:
+                    shortest = x
+                elif len(shortest) > len(x):
+                    shortest = x
+
+        if not matches:
```
Comment (qiluo-msft) [inline config/main.py:90]: I notice some code removed during backporting. Any reason?
If valid reason, why not remove on master?

Comment (jleveque) [reply config/main.py:90]: What code are you referring to that you claim was removed?

Comment (qiluo-msft) [reply config/main.py:90]: Oh I see. You removed the code on master https://github.com/Azure/sonic-utilities/pull/903/files#diff-5f0fe55ee7bf142a05fb98fce9db5b0bL95.

Then you actually backport 2 commits.

Comment (jleveque) [reply config/main.py:90]: Kinda. Actually, the other commit was backported before this one (this PR didn't have the "Request for 201911 branch" label); that's why there were conflicts and I needed to open this separately.

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Retest this please

**Comment by jleveque** [comment]
Backport #893 to the 201911 branch to support subcommand abbreviation in the `config` command

**Pull request merged**